fhir-validation/src/test/java/com/ibm/fhir/validation/test/DeviceProfileTest.java

```diff
@@ -0,0 +1,235 @@
+/*
+ * (C) Copyright IBM Corp. 2020
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.ibm.fhir.validation.test;
+
+import static com.ibm.fhir.path.util.FHIRPathUtil.compile;
+import static org.testng.Assert.assertEquals;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+import org.testng.annotations.Test;
+
+import com.ibm.fhir.model.annotation.Constraint;
+import com.ibm.fhir.model.resource.Device;
+import com.ibm.fhir.model.resource.Device.Specialization;
+import com.ibm.fhir.model.resource.OperationOutcome.Issue;
+import com.ibm.fhir.model.resource.StructureDefinition;
+import com.ibm.fhir.model.type.Canonical;
+import com.ibm.fhir.model.type.Code;
+import com.ibm.fhir.model.type.CodeableConcept;
+import com.ibm.fhir.model.type.Coding;
+import com.ibm.fhir.model.type.Extension;
+import com.ibm.fhir.model.type.Meta;
+import com.ibm.fhir.model.type.Narrative;
+import com.ibm.fhir.model.type.Uri;
+import com.ibm.fhir.model.type.Xhtml;
+import com.ibm.fhir.model.type.code.NarrativeStatus;
+import com.ibm.fhir.model.util.ValidationSupport;
+import com.ibm.fhir.profile.ConstraintGenerator;
+import com.ibm.fhir.registry.FHIRRegistry;
+import com.ibm.fhir.validation.FHIRValidator;
+import com.ibm.fhir.validation.util.FHIRValidationUtil;
+
+public class DeviceProfileTest {
+    private static final String ENGLISH_US = "en-US";
+
+    @Test
+    public void testDeviceProfile() throws Exception {
```

Comment:
I wonder if one of these should include the word MaxValueSet or ConstraintGenerator to make it clear what the test is for

Reply:
Good call.  I'll rename these to be more descriptive.